### PR TITLE
[Feature] Move db update to async method.

### PR DESCRIPTION
**Pull request by Narimm**
This migrates all the database methods that run on enable into the asynchronous task -
This includes the world caching and player caching.
This adds a callback task that runs on a time every 5 seconds until the database is completely enabled or fails...and then cancels.  The call back will log a message to the server.log saying Prism is not configured and recording yet.

Updates are now async and should not access the Bukkit API.  
DB connection is async
Db creation and setup is async.
DB updates are async
Player and worlds are cached async.

### CODE REVIEW

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity decreasing per file
==============================
+ src/main/java/me/botsko/prism/Prism.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5675143&bid=18580226)

File: src/main/java/me/botsko/prism/Prism.java [hunk 1]
```diff
@@ -326,7 +327,16 @@ public void onEnable() {
...
             pasteKey = null;
         }
+        final List<String> worldNames = getServer().getWorlds().stream()
+                .map(World::getName).collect(Collectors.toList());
+        final String[] playerNames = Bukkit.getServer().getOnlinePlayers().stream()
+                .map(Player::getName).toArray(String[]::new);
         // init db async then call back to complete enable.
+        final BukkitTask updating = Bukkit.getScheduler().runTaskTimerAsynchronously(instance, () -> {
+            if (!isEnabled()) {
+                warn("Prism is loading and updating the database logging is NOT enabled");
```
Comment (alchemistmatt) [inline src/main/java/me/botsko/prism/Prism.java]: I think a semicolon is needed here:
`Prism is loading and updating the database; logging is NOT enabled`

File: src/main/java/me/botsko/prism/Prism.java [hunk 2]
```diff
@@ -373,19 +370,32 @@ private void enabled() {
...
+            Bukkit.getScheduler().runTask(instance, () -> instance.enabled());
+            updating.cancel();
+        });
+    }
 
+    private void notifyDisabled() {
+        final String[] dbDisabled = new String[3];
+        dbDisabled[0] = "Prism will disable itself because it couldn't connect to a database.";
+        dbDisabled[1] = "If you're using MySQL, check your config. Be sure MySQL is running.";
+        dbDisabled[2] = "For help - try our Discord Channel or the Wiki on Github.";
```
Comment (alchemistmatt) [inline src/main/java/me/botsko/prism/Prism.java:394]: I would add URLs here

File: src/main/java/me/botsko/prism/Prism.java (hunk 1 above)
Comment (alchemistmatt) [reply src/main/java/me/botsko/prism/Prism.java]: I would add some blank lines before these variable declarations, but that's up to you (`worldNames`, `playerNames`, `updating`)

**Review by alchemistmatt — CHANGES_REQUESTED** [review]
Changes look fine, but I have some whitespace suggestions, plus a suggestion to add some actual website URLs

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity decreasing per file
==============================
+ src/main/java/me/botsko/prism/Prism.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5675143&bid=18580226)

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity decreasing per file
==============================
+ src/main/java/me/botsko/prism/Prism.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5675143&bid=18580226)

**Comment by Narimm** [comment]
Merge with no final review - tested on test bed before merge 
```
[08:22:56] [Server thread/INFO]: [Prism] Enabling Prism v2.1.7-SNAPSHOT-
[08:22:56] [Server thread/INFO]: [Prism] Initializing Prism 2.1.7-SNAPSHOT-. Originally by Viveleroi; maintained by the AddstarMC Network
[08:22:56] [Server thread/INFO]: [Prism] Configuring Hikari from hikari.properties
[08:22:56] [Server thread/INFO]: [Prism Debug ]: This file will not save the jdbcURL, username or password - these are loaded by default from the standard prism configuration file.  If you set these explicitly in the properties file the settings in the standard config will beignored.
[08:22:56] [Server thread/WARN]: SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
[08:22:56] [Server thread/WARN]: SLF4J: Defaulting to no-operation (NOP) logger implementation
[08:22:56] [Server thread/WARN]: SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
[08:22:56] [Server thread/INFO]: Elixr: Loaded items directory
[08:22:56] [Server thread/WARN]: [Prism] ====================================================
[08:22:56] [Server thread/WARN]: [Prism]  Prism works better if you use Paper 
[08:22:56] [Server thread/WARN]: [Prism]  as your server software. 
[08:22:56] [Server thread/WARN]: [Prism]   
[08:22:56] [Server thread/WARN]: [Prism]  Paper offers significant performance improvements,
[08:22:56] [Server thread/WARN]: [Prism]  bug fixes, security enhancements and optional
[08:22:56] [Server thread/WARN]: [Prism]  features for server owners to enhance their server.
[08:22:56] [Server thread/WARN]: [Prism]   
[08:22:56] [Server thread/WARN]: [Prism]  Paper includes Timings v2, which is significantly
[08:22:56] [Server thread/WARN]: [Prism]  better at diagnosing lag problems over v1.
[08:22:56] [Server thread/WARN]: [Prism]   
[08:22:56] [Server thread/WARN]: [Prism]  All of your plugins should still work, and the
[08:22:56] [Server thread/WARN]: [Prism]  Paper community will gladly help you fix any issues.
[08:22:56] [Server thread/WARN]: [Prism]   
[08:22:56] [Server thread/WARN]: [Prism]  Join the Paper Community @ https://papermc.io
[08:22:56] [Server thread/WARN]: [Prism] ====================================================
[08:22:56] [Server thread/INFO]: [Prism] WorldEdit not found. Certain optional features of Prism disabled.
[08:22:56] [Server thread/INFO]: [Prism] Prism bStats metrics are enabled - thank you!
[08:22:56] [Server thread/INFO]: Server permissions file permissions.yml is empty, ignoring it
[08:22:56] [Server thread/INFO]: Done (23.906s)! For help, type "help"
[08:22:56] [Craft Scheduler Thread - 0/INFO]: [Prism] Attempting to configure datasource as mysql
[08:22:56] [Craft Scheduler Thread - 0/INFO]: [Prism] Prism detected you database is version:5.7.10-log / MySQL Community Server (GPL)
[08:22:56] [Craft Scheduler Thread - 0/INFO]: [Prism] You have set nonStandardSql to true
[08:22:56] [Craft Scheduler Thread - 0/INFO]: [Prism] You are able to use non standard SQL
[08:22:57] [Craft Scheduler Thread - 0/INFO]: [Prism] Updating prism schema v7 to v8. This make take a while.
[08:22:57] [Craft Scheduler Thread - 0/INFO]: [Prism] Update check complete: Schema v8
[08:22:57] [pool-4-thread-1/INFO]: [Prism] Scheduled purge executor beginning new run...
[08:22:57] [pool-4-thread-1/INFO]: [Prism] Processed parameters for database purge: before:8w
[08:22:57] [pool-4-thread-1/INFO]: [Prism] Processed parameters for database purge: a:water-flow before:4w
[08:22:57] [pool-4-thread-1/INFO]: [Prism] Beginning prism database purge cycle. Will be performed in batches so we don't tie up the db...
[08:22:59] [Craft Scheduler Thread - 0/INFO]: [Prism] Cleared 0 rows. Max cycle time 0 msec. Using: before:8w
[08:22:59] [Craft Scheduler Thread - 0/INFO]: [Prism] Moving on to next purge rule...
[08:23:00] [Craft Scheduler Thread - 0/INFO]: [Prism] Cleared 352 rows. Max cycle time 10 msec. Using: a:water-flow before:4w
prism help
```

**Pull request merged**